connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectClusterAssertions.java

```diff
@@ -1,29 +0,0 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one or more
- * contributor license agreements. See the NOTICE file distributed with
- * this work for additional information regarding copyright ownership.
- * The ASF licenses this file to You under the Apache License, Version 2.0
- * (the "License"); you may not use this file except in compliance with
- * the License. You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package org.apache.kafka.connect.util.clusters;
-
-/**
- * @deprecated Use {@link ConnectAssertions} instead.
- */
-@Deprecated
```

Comment:
not sure whether we can remove it in 4.0 as it was deprecated in 3.7

Reply:
I'm removing it because this is not part of the public API

Reply:
Yes, it's not part of our 'official' public APIs, but I believe it's a kind of exposed testing API. Ping @gharris1727 due to your comment (https://github.com/apache/kafka/pull/14538#pullrequestreview-1674914492) regarding source compatibility.

If we all agree to remove those deprecated testing APIs, this PR can cleanup the other deprecated methods in `EmbeddedConnect` and `EmbeddedConnectCluster`

Reply:
We should remove it, the deprecation period of 1yr should be more than sufficient for an internal class.

Reply:
@gharris1727 thanks for your feedback. I will merge this PR and then create a follow-up to cleanup those deprecated internal APIs.

Reply:
https://issues.apache.org/jira/browse/KAFKA-18037